src/handlers/fedora.ts

```diff
@@ -0,0 +1,107 @@
+import { Alert } from 'react-native';
+import codePush from 'react-native-code-push';
+import {
+  // @ts-ignore
+  APP_CENTER_READ_ONLY_TOKEN_ANDROID,
+  // @ts-ignore
+  APP_CENTER_READ_ONLY_TOKEN_IOS,
+  // @ts-ignore
+  CODE_PUSH_DEPLOYMENT_KEY_ANDROID,
+  // @ts-ignore
+  CODE_PUSH_DEPLOYMENT_KEY_IOS,
+} from 'react-native-dotenv';
+import { rainbowFetch } from '../rainbow-fetch';
+import { Navigation } from '@rainbow-me/navigation';
+import Routes from '@rainbow-me/routes';
+
+const APP_CENTER_READ_ONLY_TOKEN = ios
+  ? APP_CENTER_READ_ONLY_TOKEN_IOS
+  : APP_CENTER_READ_ONLY_TOKEN_ANDROID;
+
+async function checkIfRainbowRelease(deploymentKey: string): Promise<boolean> {
+  try {
+    const response = await rainbowFetch(
+      `https://api.appcenter.ms/v0.1/apps/osdnk/rainbow-${
+        ios ? 'ios' : 'android'
+      }-codepush/deployments`,
+      {
+        headers: {
+          'X-API-Token': APP_CENTER_READ_ONLY_TOKEN,
+        },
+        method: 'get',
+      }
+    );
+    if (
+      response?.data?.find(({ key }: { key: string }) => key === deploymentKey)
+    ) {
+      return true;
+    }
+    return false;
+  } catch (e) {
+    return false;
+  }
+}
+
+export const CODE_PUSH_DEPLOYMENT_KEY = ios
+  ? CODE_PUSH_DEPLOYMENT_KEY_IOS
+  : CODE_PUSH_DEPLOYMENT_KEY_ANDROID;
+
+export const isCustomBuild = { value: false };
+
+export async function setDeploymentKey(key: string) {
+  if (!key) {
+    return;
+  }
+  const metadata = await codePush.getUpdateMetadata();
+
+  if (metadata?.deploymentKey === key) {
+    return;
+  }
+
+  const isRainbowRelease = await checkIfRainbowRelease(key);
+
+  if (!isRainbowRelease) {
+    Alert.alert(
+      'Fedora',
+      'Cannot verify the bundle! This might be a scam. Installation blocked.'
```

Comment:
Not sure if we should use i18n here

Reply:
hm, this is only for us, user will never see this so I guess there's no need